packages/aws-cdk-lib/aws-iam/lib/role.ts

```diff
@@ -551,7 +551,7 @@ export class Role extends Resource implements IRole {
     if (this._precreatedRole) {
       return this._precreatedRole.addManagedPolicy(policy);
     } else {
-      if (this.managedPolicies.find(mp => mp === policy)) { return; }
+      if (this.managedPolicies.some(mp => mp.managedPolicyArn === policy.managedPolicyArn)) { return; }
```

Comment:
We don't need an integration test, but we do need a unit test.

Reply:
I tried writing a unit test. Could you please take a look at it?

I added `addToPrincipalPolicy()` to this test because this warning only appears when `defaultPolicy` exists.